File: cpp/epi/parity.cc
#include "generic_test.h"

short Parity(unsigned long long x) {
  return 0;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::string func_name = "Parity";
  std::vector<std::string> param_names{"x"};

  return GenericTestMain(args, "parity.cc", "../test_data/epi/parity.tsv",
                         &Parity, DefaultComparator{}, func_name, param_names);
}

File: java/epi/Parity.java
package epi;

import test_framework.EpiTest;
import test_framework.GenericTest;

public class Parity {

  @EpiTest(testDataFile = "../test_data/epi/parity.tsv")
  public static short parity(long x) {
    return 0;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "Parity.java",
                new Object() {
                }.getClass().getEnclosingClass())
            .ordinal());
  }
}
